tests/integration/questionnaires.test.js

```diff
@@ -119,3 +122,151 @@ describe('QUESTIONNAIRES ROUTES - GET /questionnaires', () => {
     });
   });
 });
+
+describe('QUESTIONNAIRES ROUTES - GET /questionnaires/{_id}', () => {
+  let authToken = null;
+  beforeEach(populateDB);
+
+  describe('TRAINING_ORGANISATION_MANAGER', () => {
+    beforeEach(async () => {
+      authToken = await getToken('training_organisation_manager');
+    });
+
+    it('should get questionnaire', async () => {
+      const response = await app.inject({
+        method: 'GET',
+        url: `/questionnaires/${questionnairesList[0]._id}`,
+        headers: { Cookie: `alenvi_token=${authToken}` },
+      });
+
+      expect(response.statusCode).toBe(200);
+      expect(response.result.data.questionnaire._id).toEqual(questionnairesList[0]._id);
+    });
+
+    it('should return 404 if questionnaire does not exist', async () => {
+      const response = await app.inject({
+        method: 'GET',
+        url: `/questionnaires/${new ObjectID()}`,
+        headers: { Cookie: `alenvi_token=${authToken}` },
+      });
+
+      expect(response.statusCode).toBe(404);
+    });
+  });
+
+  describe('Other roles', () => {
+    it('should return 200 if questionnaire is published', async () => {
+      authToken = await getTokenByCredentials(noRoleNoCompany.local);
+      const response = await app.inject({
+        method: 'GET',
+        url: `/questionnaires/${questionnairesList[1]._id}`,
+        headers: { Cookie: `alenvi_token=${authToken}` },
+      });
+
+      expect(response.statusCode).toBe(200);
+    });
+
+    const roles = [
+      { name: 'helper', expectedCode: 403 },
+      { name: 'planning_referent', expectedCode: 403 },
+      { name: 'client_admin', expectedCode: 403 },
+      { name: 'trainer', expectedCode: 403 },
+    ];
+    roles.forEach((role) => {
+      it(`should return ${role.expectedCode} as user is ${role.name} and questionnaire is not published`, async () => {
```

Comment:
'and questionnaire is draft', je dirais, plutôt.
Qu'en penses-tu ?

Reply:
indeed